src/cascadia/TerminalApp/Pane.h

```diff
@@ -43,57 +43,74 @@ class Pane : public std::enable_shared_from_this<Pane>
         Horizontal = 2
     };
 
-    Pane(const GUID& profile, const winrt::Microsoft::Terminal::TerminalControl::TermControl& control, const bool lastFocused = false);
+    Pane(const GUID& profile,
+         const winrt::Microsoft::Terminal::TerminalControl::TermControl& control,
+         const bool lastFocused = false);
 
-    std::shared_ptr<Pane> GetFocusedPane();
-    winrt::Microsoft::Terminal::TerminalControl::TermControl GetFocusedTerminalControl();
+    std::shared_ptr<Pane> GetActivePane();
+    winrt::Microsoft::Terminal::TerminalControl::TermControl GetTerminalControl();
     std::optional<GUID> GetFocusedProfile();
 
     winrt::Windows::UI::Xaml::Controls::Grid GetRootElement();
 
     bool WasLastFocused() const noexcept;
-    void UpdateFocus();
+    void UpdateVisuals();
+    void ClearActive();
+    void SetActive();
 
-    void UpdateSettings(const winrt::Microsoft::Terminal::Settings::TerminalSettings& settings, const GUID& profile);
+    void UpdateSettings(const winrt::Microsoft::Terminal::Settings::TerminalSettings& settings,
+                        const GUID& profile);
     void ResizeContent(const winrt::Windows::Foundation::Size& newSize);
     bool ResizePane(const winrt::TerminalApp::Direction& direction);
     bool NavigateFocus(const winrt::TerminalApp::Direction& direction);
 
     bool CanSplit(SplitState splitType);
-    void Split(SplitState splitType, const GUID& profile, const winrt::Microsoft::Terminal::TerminalControl::TermControl& control);
+    void Split(SplitState splitType,
+               const GUID& profile,
+               const winrt::Microsoft::Terminal::TerminalControl::TermControl& control);
 
     void Close();
 
+    void SetGotFocusCallback(std::function<void(std::shared_ptr<Pane>)> pfnGotFocus);

```

Comment:
Hey, if we make this a `winrt::event`, we get lightweight delegates _and_ an easy 1:1 change when we eventually make Pane a winrt class.

Reply:
(I believe we can 100% use `winrt::event` without being ourselves a winrt class)

Reply:
Uhg. I tried so hard to get that to work before finding that the type of the event needs to be `winrt::event<winrt::delegate<std::shared_ptr<Pane>>>` ;__;

I'll fix this up, this is much better.

Reply:
I take it back, this is _not_ much better.

Right now I'm propagating the callback down to the leaves, then back up to the parent when leaves are closed.

With a `winrt::event`, I'm having a much harder time getting this right. Having the Tab register a `GotFocus` handler on the panes works just fine, but when the panes are closed, they just _will not_ activate the right pane in the tree. I'll keep playing with it.